cmd/download_headers.cpp

```diff
@@ -88,7 +89,9 @@ int main(int argc, char* argv[]) {
         // Sentry client - connects to sentry
         SentryClient sentry{sentry_addr};
         sentry.set_status(head_hash, head_td, chain_identity);
+        sentry.hand_shake();
         message_receiving = std::thread([&sentry]() { sentry.execution_loop(); });
```

Comment:
There is a problem with thread and exceptions.
For instance `sentry.execution_loop()` and it's cascading calls might throw exceptions but there is no exception handler in its own thread (the try catch is in the main thread but there is no automatic propagation of exceptions among threads) so the program terminates abruptly.
Either ensure all call stacks in the thread are no except (i.e. use return codes) or follow the pattern in `worker.hpp` to capture foreign threads exceptions.

Reply:
Yes, I know, it is a little messy.
Do you know the requirement that stage-sync loop has upon each stage? I prefer to spend time to correctly handle program exit when all the final requirement will be known.